### PR TITLE
kustomize: migrate presubmits to community clusters

### PR DESCRIPTION
**Pull request by ShivamTyagi12345**
This PR transitions the `kustomize-presubmit-master` job from the default cluster to `eks-prow-build-cluster` :+1: 

ref: https://github.com/kubernetes/test-infra/issues/29722

### CODE REVIEW

**Comment by natasha41575** [comment]
@ShivamTyagi12345 I don't know anything about the migration or prow infrastructure, maybe you can reach out to someone there to figure out how to get the test to pass?

**Comment by rjsadow** [comment]
xslack: https://kubernetes.slack.com/archives/C09QZ4DQB/p1686255333214949

The test is failing because `spec.container.resources` is blank. You can see the complaints about the failing job [Here](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/29733/pull-test-infra-unit-test/1666896907416375296). This is because the GKE clusters don't require resource requests/limits to be set whereas the community owned clusters do.

I would recommend setting some default values (you can ref [#29629](https://github.com/kubernetes/test-infra/pull/29724/files)). Then reach out to the [Kustomize OWNERS](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/kustomize/OWNERS) to see if they have any recommendations. If they're not sure then you can merge with the default values and iterate to fine tune the requests and limits if the jobs start failing.

File: config/jobs/kubernetes-sigs/kustomize/kustomize-presubmit-master.yaml [hunk 1]
```diff
@@ -15,7 +16,11 @@ presubmits:
         - prow-presubmit-check
         resources:
           requests:
+            memory: "2000Mi"
             cpu: 4000m
+          limits:
+            memory: "2000Mi"
+            cpu: 500m
```
Comment (rjsadow) [inline config/jobs/kubernetes-sigs/kustomize/kustomize-presubmit-master.yaml:23]: This will need to be 4000m to match the request

Comment (ShivamTyagi12345) [reply config/jobs/kubernetes-sigs/kustomize/kustomize-presubmit-master.yaml:23]: I am doing this change. Let's look into the results

**Comment by ShivamTyagi12345** [comment]
@rjsadow I see these values lead to the similar error

File: config/jobs/kubernetes-sigs/kustomize/kustomize-presubmit-master.yaml [hunk 2]
```diff
@@ -15,7 +16,11 @@ presubmits:
         - prow-presubmit-check
         resources:
           requests:
+            memory: "2000Mi"
             cpu: 4000m
+          limits:
+            memory: "2000Mi"
+            cpu: 4000m
```
Comment (rjsadow) [inline config/jobs/kubernetes-sigs/kustomize/kustomize-presubmit-master.yaml:23]: The linter is erroring on the trailing white space after this line.

`          cpu: 4000m                    ` 

```
config/jobs/kubernetes-sigs/kustomize/kustomize-presubmit-master.yaml
  23:23     error    trailing spaces  (trailing-spaces) 
  ```

Comment (ShivamTyagi12345) [reply config/jobs/kubernetes-sigs/kustomize/kustomize-presubmit-master.yaml]: Yes I have remove d the trailing space, How are you seeing that this was the error?

Comment (rjsadow) [reply config/jobs/kubernetes-sigs/kustomize/kustomize-presubmit-master.yaml]: It looks like there is still some trailing whitespace after the CPU. You can look at the log that the k8s-ci-robot provides or from the job itself. Here's the link to the job logs https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/29733/pull-test-infra-verify-lint/1669558900354977792. In there if you just search for "error" you'll find the offending line(s).

**Comment by ameukam** [comment]
/retitle kustomize: migrate presubmits to community clusters

**Pull request merged**